File: katas/5 kyu/rgb-to-hex-conversion/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def rgb(r, g, b):
  r = check(r)
  b = check(b)
  g = check(g)
  return ('%02X%02X%02X' % (r, g, b))

def check(int):
  if int >= 255:
    return 255
  if int <= 0:
    return 0
  else :
    return int
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def rgb(r, g, b):
    round = lambda x: min(255, max(x, 0))
    return ("{:02X}" * 3).format(round(r), round(g), round(b))
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(rgb(0,0,0),"000000", "testing zero values")
test.assert_equals(rgb(1,2,3),"010203", "testing near zero values")
test.assert_equals(rgb(255,255,255), "FFFFFF", "testing max values")
test.assert_equals(rgb(254,253,252), "FEFDFC", "testing near max values")
test.assert_equals(rgb(-20,275,125), "00FF7D", "testing out of range values")

File: katas/5 kyu/rgb-to-hex-conversion/solution.php
<!-- rgb-to-hex-conversion -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function rgb($r,$g,$b){
  var_dump("Debugging:", ['r'=>$r, 'g'=>$g, 'b'=>$b]);
  return sprintf("%02X%02X%02X", chk($r), chk($g), chk($b));
}
function chk ($n) { return $n > 255 ? 255 : ($n < 0 ?  0 : $n);}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase
{
    public function testBaseTests() {
    // assertEquals(mixed $expected, mixed $actual[, string $message = ''])
      $this->assertSame("FFFFFF", rgb(255, 255, 255));
      $this->assertSame("FFFFFF", rgb(255, 255, 300));
      $this->assertSame("000000", rgb(0,0,0));
      $this->assertSame("000000", rgb(-500,0,0));
      $this->assertSame("9400D3", rgb(148, 0, 211));
    }
}